Mage.Sets/src/mage/cards/g/GerrardsHourglassPendant.java

```diff
@@ -0,0 +1,129 @@
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.common.ExileSourceCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.keyword.FlashAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterPermanentCard;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.card.PutIntoGraveFromBattlefieldThisTurnPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.watchers.common.CardsPutIntoGraveyardWatcher;
+
+import java.util.UUID;
+
+public final class GerrardsHourglassPendant extends CardImpl {
+
+    public GerrardsHourglassPendant(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{1}");
+        addSuperType(SuperType.LEGENDARY);
+
+        // Flash
+        this.addAbility(FlashAbility.getInstance());
+
+        // If a player would begin an extra turn, that player skips that turn instead.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GerrardsHourglassPendantSkipExtraTurnsEffect()));
+
+        // {4}, {T}, Exile Gerrard's Hourglass Pendant: Return to the battlefield tapped all artifact, creature,
+        // enchantment, and land cards in your graveyard that were put there from the battlefield this turn.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new GerrardsHourglassPendantReanimateEffect(), new GenericManaCost(4));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new ExileSourceCost());
+        ability.addWatcher(new CardsPutIntoGraveyardWatcher());
+        this.addAbility(ability);
+    }
+
+    private GerrardsHourglassPendant(final GerrardsHourglassPendant card) {
+        super(card);
+    }
+
+    @Override
+    public GerrardsHourglassPendant copy() {
+        return new GerrardsHourglassPendant(this);
+    }
+}
+
+class GerrardsHourglassPendantSkipExtraTurnsEffect extends ReplacementEffectImpl {
+
+    public GerrardsHourglassPendantSkipExtraTurnsEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Detriment);
+        staticText = "If a player would begin an extra turn, that player skips that turn instead";
+    }
+
+    public GerrardsHourglassPendantSkipExtraTurnsEffect(final GerrardsHourglassPendantSkipExtraTurnsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public GerrardsHourglassPendantSkipExtraTurnsEffect copy() {
+        return new GerrardsHourglassPendantSkipExtraTurnsEffect(this);
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        Player player = game.getPlayer(event.getPlayerId());
```

Comment:
If the player is null, doesn't that mean that there's no extra turn to skip?

Reply:
Yes - do you think that the `player != null` condition should be taken out? This part of the implementation is based on [Ugin's Nexus](https://github.com/magefree/mage/blob/80e11b20528dfb88a91c3f3be66a354818a0606b/Mage.Sets/src/mage/cards/u/UginsNexus.java) in which I assumed it was just a failsafe

Reply:
Actually I am not so sure of my first comment. I don't know what happens to any extra turns a player has if they lose the game. (Or what happens to the rest of a player's regular turn if they lose the game).

Yes, generally old cards are a good starting point, but iirc extra turns are still a little buggy in XMage. I also cannot imagine that extra turns show up very often so are not tested as often, and I imagine that's doubly so about effects which make players skip them.

Reply:
Just tested giving a player an extra turn with Time Warp then making them lose the game with Door to Nothingness and there was no misinteraction with the extra turn e.g. no message in chat to indicate that an extra turn began. As in the [other review](https://github.com/magefree/mage/pull/9513#pullrequestreview-1103443452) I think the `player != null` condition is just supposed to be for good practice before attempting to call `player.getLogName()` as opposed to a necessity for edge cases